govcd/nsxv_ipset.go

```diff
@@ -0,0 +1,254 @@
+/*
+ * Copyright 2019 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+
+package govcd
+
+import (
+	"encoding/xml"
+	"fmt"
+	"net/http"
+
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	"github.com/vmware/go-vcloud-director/v2/util"
+)
+
+func (vdc *Vdc) CreateNsxvIpSet(ipSetConfig *types.EdgeIpSet) (*types.EdgeIpSet, error) {
+	if err := validateCreateNsxvIpSet(ipSetConfig); err != nil {
+		return nil, err
+	}
+
+	vdcId, err := getUuidFromHref(vdc.Vdc.HREF)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get vdc ID from HREF: %s", err)
+	}
+
+	// build a path for IP set creation. The endpoint should look like:
+	// https://_hostname_/network/services/ipset/f9daf2da-b4f9-4921-a2f4-d77a943a381c where the
+	// trailing UUID is vDC ID
+	httpPath, err := vdc.buildNsxvNetworkServiceEndpointURL(types.NsxvIpSetServicePath + "/" + vdcId)
+	if err != nil {
+		return nil, fmt.Errorf("could not get network services API endpoint for IP set: %s", err)
+	}
+
+	// Success or an error of type types.NSXError is expected
+	_, err = vdc.client.ExecuteParamRequestWithCustomError(httpPath, nil, http.MethodPost, types.AnyXMLMime,
+		"error creating IP set: %s", ipSetConfig, &types.NSXError{})
+	if err != nil {
+		return nil, err
```

Comment:
Custom message on what action error happen would be helpfull

Reply:
error message in line 34